### PR TITLE
Add M2 drives to Gimlet thermal control loop

### PR DESCRIPTION
**Pull request by mkeeter**

### CODE REVIEW

File: task/thermal/src/bsp/gimlet_b.rs [hunk 1]
```diff
@@ -74,19 +83,43 @@ impl Bsp {
...
+                let m = drv_i2c_devices::max5970::Max5970::new(&dev);
+                let mut out = POWER_STATE_A0;
+                match m.read_reg(drv_i2c_devices::max5970::Register::status3) {
+                    Ok(s) => {
+                        // pg[0]
+                        if s & 1 != 0 {
+                            out |= POWER_M2A;
+                        }
+                        // pg[1]
+                        if s & 2 != 0 {
```
Comment (jgallagher) [inline task/thermal/src/bsp/gimlet_b.rs:110]: Tiny nitpick - `s & (1 << 0)` and `s & (1 << 1)` might be slightly clearer

File: task/thermal/src/bsp/gimlet_b.rs [hunk 2]
```diff
@@ -74,19 +83,43 @@ impl Bsp {
...
+                            out |= POWER_M2B;
+                        }
+                    }
+                    // TODO: error handling here?
```
Comment (jgallagher) [inline task/thermal/src/bsp/gimlet_b.rs:114]: Do we need any error handling? Locally it looks like if we fail to read this register for any reason we have to assume we can't access the M.2 devices, unless there are specific error codes we might be able to do something with (retry?)?

**Review by jgallagher — APPROVED** [review]
LGTM, just a couple possible clarity suggestions

File: task/thermal/src/bsp/gimlet_b.rs [hunk 3]
```diff
@@ -52,13 +52,22 @@ pub(crate) struct Bsp {
...
+    i2c_task: TaskId,
+
     /// Tuning for the PID controller
     pub pid_config: PidConfig,
 }
 
 // Use bitmasks to determine when sensors should be polled
-const POWER_STATE_A2: u32 = 0b001;
-const POWER_STATE_A0: u32 = 0b010;
+const POWER_STATE_A2: u32 = 0b0001;
```
[the remark this answers is not in the document]
Comment (mkeeter) [reply task/thermal/src/bsp/gimlet_b.rs]: Good idea, I converted to `bitflags!` and am retesting now

**Pull request merged**